src/test/groovy/graphql/language/PrettyPrinterTest.groovy

```diff
@@ -0,0 +1,529 @@
+ # before def #2
+scalar 
+MyScalar # def trailing 
+# after def
+'''
+
+        def expected = '''# before def #1
+"""
+ Description 
+"""
+# before def #2
+scalar MyScalar # def trailing 
+# after def
+'''
+
+        when:
+        def result = print(input)
+        then:
+
+        result == expected
+    }
+
+
+    def "can print comments in directives"() {
+        given:
+        def input = '''
+# before def #1
+""" Description def """
+ # before def #2
+directive 
+@myDirective( # inside block #1
+    # arg1 #1
+   """ Description arg1 """
+   
+    # arg1 #2
+   arg1: String!      # arg1 trailing
+   # arg2 #1
+  arg2: ID arg3: String! # arg3 trailing
+  # inside block #1
+ ) on FIELD_DEFINITION # trailing def
+# after def
+'''
+
+        def expected = '''# before def #1
+"""
+ Description def 
+"""
+# before def #2
+directive @myDirective( # inside block #1
+  # arg1 #1
+  """
+   Description arg1 
+  """
+  # arg1 #2
+  arg1: String! # arg1 trailing
+  # arg2 #1
+  arg2: ID
+  arg3: String! # arg3 trailing
+  # inside block #1
+) on FIELD_DEFINITION # trailing def
+# after def
+'''
+
+        when:
+        def result = print(input)
+        then:
+
+        result == expected
+    }
+
+    def "can print extended type with comments"() {
+        given:
+        def input = '''
+# before description
+extend type Query { # beginning of block
+a: A b: B
+        # end of block inside #1
+# end of block inside #2
+}               # end of block
+'''
+
+        def expected = '''# before description
+extend type Query { # beginning of block
+  a: A
+  b: B
+  # end of block inside #1
+  # end of block inside #2
+} # end of block
+'''
+        when:
+        def result = print(input)
+
+        then:
+        result == expected
+    }
+
+
+    def "can print input type with comments"() {
+        given:
+        def input = '''
+# before description
+""" Description """
+# before def #1
+# before def #2
+input MyInput { # beginning of block
+a: A b: B
+        # end of block inside #1
+# end of block inside #2
+}               # end of block
+'''
+
+        def expected = '''# before description
+"""
+ Description 
+"""
+# before def #1
+# before def #2
+input MyInput { # beginning of block
+  a: A
+  b: B
+  # end of block inside #1
+  # end of block inside #2
+} # end of block
+'''
+        when:
+        def result = print(input)
+
+        then:
+        result == expected
+    }
+
+    def "can use print with tab indent"() {
+        given:
+        def input = '''
+type Query {
+field(
+# comment
+a: A, b: B): Type
+}
+'''
+
+        def expected = '''type Query {
+\tfield(
+\t\t# comment
+\t\ta: A
+\t\tb: B
+\t): Type
+}
+'''
+        when:
+        def options = PrettyPrinter.PrettyPrinterOptions
+                .builder()
+                .indentType(PrettyPrinter.PrettyPrinterOptions.IndentType.TAB)
+                .indentWith(1)
+                .build()
+
+        def parser = new NodeToRuleCapturingParser()
```

Comment:
I think it would be good that the pretty printer would be self contained and not requre the user to parse anything themself.

Reply:
It actually is. I created this static `print` method that callers can use just passing the schemaDef and options:
```
    public static String print(String schemaDefinition, PrettyPrinterOptions options) {
        NodeToRuleCapturingParser parser = new NodeToRuleCapturingParser();
        Document document = parser.parseDocument(schemaDefinition);

        return new PrettyPrinter(parser.getParserContext(), options).print(document);
    }
```
I had created the tests before creating the method, and never got around to change the call in the tests :-)

Reply:
can reflect this in the test then please